consensus/consensus-types/src/block.rs

```diff
@@ -237,133 +156,67 @@ where
             ),
         );
 
-        let block_internal = BlockSerializer::<T> {
-            payload: None,
-            epoch: ledger_info.epoch() + 1,
-            round: 0,
-            timestamp_usecs: ledger_info.timestamp_usecs(),
-            quorum_cert: &genesis_quorum_cert,
-            author: None,
-        };
+        let block_data = BlockData::new_genesis(ledger_info.timestamp_usecs(), genesis_quorum_cert);
 
         Block {
-            id: block_internal.hash(),
-            epoch: ledger_info.epoch() + 1,
-            round: 0,
-            timestamp_usecs: ledger_info.timestamp_usecs(),
-            quorum_cert: genesis_quorum_cert,
-            block_type: BlockType::Genesis,
+            id: block_data.hash(),
+            block_data,
+            signature: None,
         }
     }
 
-    // Create a block directly.  Most users should prefer make_block() as it ensures correct block
-    // chaining.  This functionality should typically only be used for testing.
-    pub fn new_internal(
-        payload: T,
-        epoch: u64,
-        round: Round,
-        timestamp_usecs: u64,
-        quorum_cert: QuorumCert,
-        validator_signer: &ValidatorSigner,
-    ) -> Self {
-        let author = validator_signer.author();
-        let block_internal = BlockSerializer {
-            payload: Some(&payload),
-            epoch,
-            round,
-            timestamp_usecs,
-            quorum_cert: &quorum_cert,
-            author: Some(&author),
-        };
-
-        let id = block_internal.hash();
-        let signature = validator_signer
-            .sign_message(id)
-            .expect("Failed to sign message");
+    /// The NIL blocks are special: they're not carrying any real payload and are generated
+    /// independently by different validators just to fill in the round with some QC.
+    pub fn new_nil(round: Round, quorum_cert: QuorumCert) -> Self {
+        let block_data = BlockData::new_nil(round, quorum_cert);
 
         Block {
-            id,
-            epoch,
-            round,
-            timestamp_usecs,
-            quorum_cert,
-            block_type: BlockType::Proposal {
-                payload,
-                author,
-                signature: signature.into(),
-            },
+            id: block_data.hash(),
+            block_data,
+            signature: None,
         }
     }
 
-    pub fn make_block(
-        parent_block: &Block<T>,
+    pub fn new_proposal(
```

Comment:
I actually liked `make_block` more: we're generating blocks in different scenarios, e.g., retrieving the blocks from remote peers, after restart, etc., `new_proposal` sounds weird for these cases.